### ISSUE TITLE
Should show GNOME wallpapers too, if present

### ISSUE THREAD

**Issue by jbicha** [mate-desktop/mate-control-center#268]
Impact
======
`mate-appearance-properties` only looks for backgrounds that register themselves in /[usr/share/mate-background-properties/](url)

The only packages in Ubuntu that use their directory are `mate-backgrounds`, `mate-screensavers-common `and the `ubuntu-mate-wallpapers-*` packages.

Quite a few packages use` /usr/share/gnome-background-properties` instead.

Test Case
=========
Install `ubuntu-wallpapers` (If you're not using a Ubuntu-derived system, install `gnome-backgrounds` instead.)
Right-click on the MATE desktop and select Change Desktop Background
The Ubuntu default wallpapers should show in the list.

Suggested Fix
===========
1. I recommend that MATE simply switch to using the standard `gnome-background-properties directory`.

1.b. Or MATE could do that but still show wallpapers from the old `mate-background-properties` directory.

2. Or MATE could keep its wallpapers packages as is and also search in the `gnome-background-properties` directory. If MATE wants its wallpapers to show in GNOME, then it needs to convince the gnome-control-center developers to accept a patch to show wallpapers from the `mate-background-properties` directory.

**Comment by raveit65** [comment]
Main wallpaper dir is /usr/share/backgrounds which is used by all De's in fedora.
And i see all wallpapers from there in mate-appearance-properties.
Also mate-backgrounds use that dir.

**Comment by raveit65** [comment]
This helps!
```
[root@mother mate-background-properties]# ln -s /usr/share/gnome-background-properties/adwaita.xml 
[root@mother mate-background-properties]# ln -s /usr/share/gnome-background-properties/gnome-backgrounds.xml 
```
Those xml files are used to register their wallpapers from
/usr/share/backgrounds/gnome/
But distros like fedora ships a beautiful package '${release}-backgrounds', which ships xml files for the fedora default wallpaper for all DEs.
In boths dirs you will find the f24.xml .
```
[root@mother mate-background-properties]# ls
f22-extras.xml  f24.xml    mate-desktop.xml
cosmos.xml      f23-extras.xml  mate-nature.xml
f21-extras.xml  f24-extras.xml  mate-abstract.xml
[root@mother mate-background-properties]# ls /usr/share/gnome-background-properties/
adwaita.xml  f24.xml  gnome-backgrounds.xml
```
So changing the dir to /usr/share/gnome-background-properties/ is a bad idea for fedora or maybe others, as the mate dir is used by other fedora packages.
Maybe mate-appearance-properties can check for both dirs.
Than only the fedora default is listed twice.
Or you file out a report at ubuntu-mate and ask the maintainer there to ship those symlinks ;-)

**Comment by jbicha** [comment]
I disagree that it's a bad idea. The only reason some distros ship some wallpapers in the MATE directory is because they want their wallpapers to work in MATE and they can easily stop doing that. Most wallpaper packages don't register themselves in `mate-background-properties`.

There's no benefit to having a `mate-background-properties` directory. The only reason it exists is because of the copy/replace done when MATE first started and not because of a conscious decision that it was a good idea.

**Comment by clefebvre** [comment]
Similar report on Cinnamon: https://github.com/mate-desktop/mate-control-center/issues/268.

I closed it so we could all discuss here rather than in both places (since we're talking about the very same thing).

**Comment by jbicha** [comment]
For clarification, the previous link should be https://github.com/linuxmint/Cinnamon/issues/6289

**Comment by clefebvre** [comment]
The way things work is the same in GNOME, MATE and Cinnamon; i.e. the backgrounds can be placed anywhere on the file systems, they're loaded by metadata.

There are multiple advantages of doing that over just loading whatever is placed within some directory:

- It allows to ship different sets of backgrounds for different DEs.
- It allows to group sets of backgrounds in different places on the filesystem (and thus not to duplicate them)
- It allows DEs to control their own metadata, for instance Cinnamon supports artists tags
- It allows DEs to group backgrounds by metadata sets, this is done by Cinnamon for instance, where each metadata definition provides a distinct collection.

> There's no benefit to having a mate-background-properties directory. The only reason it exists is because of the copy/replace done when MATE first started and not because of a conscious decision that it was a good idea.

Switching to /usr/share/cinnamon-background-proprerties was a very conscious decision and it was done with purpose. Other than the reasons I gave above, it allowed us to ship a clean DE without being impacted by some of the poor packaging done in Debian where maintainers shoved artwork in dependencies which couldn't be removed or sometimes even within core packages.

We've always also insisted for MATE not to depend on its artwork, so that distributions can either decide to ship MATE artwork or their own artwork instead.

We don't provide artwork with Cinnamon, and if we do in the future, we'll respect the fact that people might not want it and so it will be optional.

Personally, I don't want to see GNOME, MATE, Debian, or even KDE Plasma (why not after all) branding in Cinnamon unless it was meant to be there.

**Comment by clefebvre** [comment]
The solution imo is to fix the background packages themselves. If all they do is define backgrounds for GNOME then they only work in GNOME, so maybe they should reflect that in their name.

They can easily support MATE and Cinnamon by providing additional metadata (and I hope they'll support the artist tag if they support Cinnamon).

They should also support Xfce by placing links in /usr/share/xfce4/backdrops/.

and they should also support KDE which is also quite popular, by placing both bgs and metas in KDE format in /usr/share/wallpapers/.

**Comment by clefebvre** [comment]
Artwork provided by the DEs themselves should not be "fixed" though. I don't mate-backgrounds should show in other DEs, I don't think I should see Breeze backgrounds in Cinnamon just because I or another local user installed KDE, etc etc..

**Comment by raveit65** [comment]
> I disagree that it's a bad idea. The only reason some distros ship some wallpapers in the MATE directory is because they want their wallpapers to work in MATE and they can easily stop doing that. Most wallpaper packages don't register themselves in `mate-background-properties`.

Sadly, that this isn't usable in fedora seems no reason for you.
And again, this can be easily fixed by your distro with providing symlinks.
